pkg/tlsutil/tls.go

```diff
@@ -153,19 +157,59 @@ func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *v1.Service
 		return nil, nil, nil, err
 	}
 	caSecretAndConfigMapName := ToCASecretAndConfigMapName(k, n)
-	caSecret, caConfigMap, err := getCASecretAndConfigMapInCluster(scg.KubeClient, caSecretAndConfigMapName, ns)
+
+	var (
+		caSecret    *v1.Secret
+		caConfigMap *v1.ConfigMap
+	)
+
+	caSecret, caConfigMap, err = getCASecretAndConfigMapInCluster(scg.KubeClient, caSecretAndConfigMapName, ns)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
+	if config.CAKey != "" && config.CACert != "" {
```

Comment:
We can have a boolean value `hasPassedInCA := config.CAKey != "" && config.CACert != ""`.
And we only call `getCASecretAndConfigMapInCluster` if `hasPassedInCA` is false. e.g
```go
if hasPassedInCA {
...
}

if config.CAKey != "" || config.CACert != "" {
	// if only one of the custom CA Key or Cert is provided
	return nil, nil, nil, ErrCAKeyAndCACertReq
}

if !hasPassedInCA {
  caSecret, caConfigMap, err = getCASecretAndConfigMapInCluster(scg.KubeClient, caSecretAndConfigMapName, ns)
 if err != nil {
   return nil, nil, nil, err
 }
}


```

Reply:
It doesnt really matter cause if `getCASecretAndConfigMapInCluster` is called before and no configMap exists the values of  caSecret & caConfigMap will be nil. Then in the ` if config.CAKey != "" && config.CACert != "" {` block the values get updated

Reply:
@rithujohn191 your code is correct. My suggestion is just an optimization because `getCASecretAndConfigMapInCluster` makes calls to k8s cluster which is unnecessary if custom ca is passed.

Reply:
I had the same thought process, I think code readability is better the way it is right now and 2 GET calls from the api server is not heavy weight at all. I think the way the code is currently written  makes it more clear that the custom CA takes preference. But if you insist I could change it. Either way its not much of a difference.

Reply:
@rithujohn191 let's just keep the code as it is for readability.  we can revisit the code if we want to optimize it.